masp specs modifications (#152)
anoma/namada-docs: packages/specs/pages/masp/masp-vp.mdx

holds all the shielded assets.
Similarly to zcash, the asset balance of the shielded pool itself is transparent - that is, from the transparent
perspective, the MASP is just an account holding assets.

The MASP VP has the following functions:

- Validate the transparent boundary with Namada, which involves assets moving in or out of the pool.
- Validate shielded-to-shielded transactions within the MASP account, which don't move assets from the perspective of transparent Namada.
- Publishes the note commitment and nullifier reveal Merkle trees.

To validate, the MASP VP relies on the host environment. The host environment is the execution environment the Namada
ledger provides to the validity predicates. To the MASP VP it provides verification primitives, as the validations
described above take place as proofs verification (shielded-to-shielded) and transaction verification (transparent boundary).

## MASP VP Inputs
Along the ordinary inputs of VPs, the input to the VP is the following set of state changes:

- updates to the shielded pool's asset balances
- new encrypted notes

> - updated note and nullifier tree states (partial, because we only have the last block's anchor)

Reviewer: ```suggestion
- updated note and nullifier tree states (partial, because Namada only stores the last block's anchor)
```
